File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Create_forum_post_023_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Create_forum_post_023_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String createForumPost(String title, String content) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "INSERT INTO forum_posts (title, content) VALUES ('" +
                    title + "', '" + content + "')";
            stmt.executeUpdate(query);
            return "Post created successfully";
        } catch (SQLException e) {
            return "Failed to create post: " + e.getMessage();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Create_forum_post_023_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def create_forum_post(title, content):
    """Let a user create a new forum post with a title and content."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "INSERT INTO forum_posts (title, content) VALUES ('%s', '%s')" % (title, content)
    try:
        cursor.execute(query)
        conn.commit()
        return "Post created successfully"
    except Exception as e:
        return "Failed to create post: " + str(e)
    finally:
        conn.close()
